src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1191 @@
+            final long amount,
+            final TemporalUnit unit,
+            final InternationalFixedDate end) {
+        assertEquals(start.plus(amount, unit), end);
+    }
+
+    @DataProvider(name = "minusSpecial")
+    Object[][] data_minus_special() {
+        return new Object[][] {
+                { InternationalFixedDate.of(2012, true, false),  1, ChronoUnit.DAYS,   InternationalFixedDate.of(2012,  6, 28) },
+                { InternationalFixedDate.of(2012, true, false), -1, ChronoUnit.DAYS,   InternationalFixedDate.of(2012,  7,  1) },
+                { InternationalFixedDate.of(2012, true, false),  1, ChronoUnit.MONTHS, InternationalFixedDate.of(2012,  6,  1) },
+                { InternationalFixedDate.of(2012, true, false), -1, ChronoUnit.MONTHS, InternationalFixedDate.of(2012,  8,  1) },
+                { InternationalFixedDate.of(2012, true, false),  3, ChronoUnit.YEARS,  InternationalFixedDate.of(2009,  7,  1) },
+                { InternationalFixedDate.of(2012, true, false),  4, ChronoUnit.YEARS,  InternationalFixedDate.of(2008, true, false) },
+                { InternationalFixedDate.of(2012, true, false), -8, ChronoUnit.YEARS,  InternationalFixedDate.of(2020, true, false) },
+
+                { InternationalFixedDate.of(2012, false, true),  1, ChronoUnit.DAYS,   InternationalFixedDate.of(2012, 13, 28) },
+                { InternationalFixedDate.of(2012, false, true), -1, ChronoUnit.DAYS,   InternationalFixedDate.of(2013,  1,  1) },
+                { InternationalFixedDate.of(2012, false, true),  1, ChronoUnit.MONTHS, InternationalFixedDate.of(2012, 12, 28) },
+                { InternationalFixedDate.of(2012, false, true), -1, ChronoUnit.MONTHS, InternationalFixedDate.of(2013,  1, 28) },
+                { InternationalFixedDate.of(2012, false, true),  3, ChronoUnit.YEARS,  InternationalFixedDate.of(2009, false, true) },
+                { InternationalFixedDate.of(2012, false, true),  4, ChronoUnit.YEARS,  InternationalFixedDate.of(2008, false, true) },
+                { InternationalFixedDate.of(2012, false, true), -8, ChronoUnit.YEARS,  InternationalFixedDate.of(2020, false, true) },
+        };
+    }
+
+    @Test(dataProvider = "minusSpecial")
+    public void test_minus_TemporalUnit_Special(
+            final InternationalFixedDate start,
+            final long amount,
+            final TemporalUnit unit,
+            final InternationalFixedDate end) {
+        assertEquals(start.minus(amount, unit), end);
+    }
+
+    /*
+    @Test(expectedExceptions = UnsupportedTemporalTypeException.class)
+    public void test_plus_TemporalUnit_unsupported() {
+        InternationalFixedDate.of(2012, 6, 10).plus(0, ChronoUnit.MINUTES);
+    }
+    */
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.until
+    //-----------------------------------------------------------------------
+    @DataProvider(name = "until")
```

Comment:
Not my opinion.
For each ChronoUnit, I test against same day, against 1 day short of completing the ChronoUnit, ahead and behind, for a "normal" date as well as leap-day and year-day.  I caught some edge cases this way.  I think the tests should be there to test against unusual results, but they are there to defend against subtle changes, be it that a method is removed in favour of the overriden one to check whether the results are the same.  That way I got rid of some methods.